Migrate default billing addresses to address book

ikraamg opened the pull request:
Edit: Specs have now been added and an upgrade task is now used, as it was previously used in [Solidus 1.3](https://github.com/solidusio/solidus/blob/v2.3/core/lib/tasks/upgrade.rake)

**Description**
`Spree::User` records have a `bill_address_id` that denotes their saved
billing address to be reused in the next checkout. That column, however,
is not in use since Solidus 2.11.0. Instead, we rely on a`default_billing`
Boolean on the `spree_user_addresses` join table.

This migration sets the `default_billing` flag on that join table for
addresses matching the user's default billing address and the user's ID.

For stores that have already started using the `default_billing`
Boolean, this task excludes the user_addresses that are already using
the new format by setting those `bill_address_id`s to `nil` before
running the query.


<!--
  Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

  Please include screenshots in case of visual changes to the frontend or backend sections of Solidus.

  If needed you can reference another PR or issue here, e.g.:
  Ref #ISSUE
-->

**Checklist:**
- [ ] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [ ] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [ ] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)

DanielePalombo commented:
Nice PR, please add some test on it.

About the `upgrade` task, probably it will be a big changes for this PR, we might propose to @kennyadsl to add this gem https://github.com/ilyakatz/data-migrate .

ikraamg committed:
Migrate default billing addresses to address book

`Spree::User` records have a `bill_address_id` that denotes their saved
billing address to be reused in the next checkout. That column, however,
is not in use since Solidus 2.11.0. Instead, we rely on a`default_billing`
Boolean on the `spree_user_addresses` join table.

This migration sets the `default_billing` flag on that join table for
addresses matching the user's default billing address and the user's ID.

For stores that have already started using the `default_billing`
Boolean, this task excludes the user_addresses that are already using
the new format by setting those `bill_address_id`s to `nil` before
running the query.

Co-authored-by: mamhoff <mamhoff@gmail.com>
Co-authored-by: DanielePalombo <danielepalombo@nebulab.it>

ikraamg commented:
Thanks, @DanielePalombo.

Specs have now been added and an upgrade task is now used again, as it was previously used in [Solidus 1.3](https://github.com/solidusio/solidus/blob/v2.3/core/lib/tasks/upgrade.rake)

Please let me know if you have any additional suggestions.

mamhoff commented:
Thank you @ikraamg @DanielePalombo !
